app/serializers/user_bookmark_serializer.rb

```diff
@@ -83,7 +84,11 @@ def closed
   end
 
   def highest_post_number
-    topic.highest_post_number
+    scope.user.staff? ? topic.highest_staff_post_number : topic.highest_post_number
+  end
+
+  def last_read_post_number
+    topic.topic_users.find { |tu| tu.user_id == scope.user.id }&.last_read_post_number
```

Comment:
Just want to confirm that `topic_users` have been eager loaded? Otherwise, this is going to cause an N+1 queries problem.

Reply:
Also, I wonder if we need to load all the `topic_users` of a topic when the bookmark is already scoped to a single user. Feels like the fetching of this data can be optimized further.

Reply:
Sorry XD Just read `bookmark_query.rb`.